app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,98 @@
+class PostsController < ApplicationController
+
+  before_action :authenticate_user!, except: [:index, :show]
+  before_action :find_post, only: [:show, :edit, :update, :destroy]
+  before_action :authorize_post, only: [:edit, :update, :destroy]
+
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+
+    # important association
+    @post.user = current_user
+
+    if @post.save
+      #render text: "SUCCESS"
+      redirect_to post_path(@post), notice: "Post created"
+    else
+      #render text: "FAILURE"
+      flash[:alert] = "Post not created"
```

Comment:
Since you have the counting of error message and list of error. Maybe it is not necessary to have this alert?